x/go/stack/stack.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2023 Synnax Labs, Inc.
+//
+// Use of this software is governed by the Business Source License included in the file
+// licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with the Business Source
+// License, use of this software will be governed by the Apache License, Version 2.0,
+// included in the file licenses/APL.txt.
+
+package stack
+
+import (
+	"errors"
+)
+
+type Stack[T any] struct {
```

Comment:
For generic, STL like containers, it's a good idea to add tests and to document.